### PR TITLE
Add optional defmt support for error types

### PR DESCRIPTION
**Pull request by vhdirk**
add defmt as an optional dependent package

closes https://github.com/rust-embedded-community/serde-json-core/issues/75
closes #76

### CODE REVIEW

File: src/de/mod.rs [hunk 1]
```diff
@@ -74,6 +75,7 @@ pub enum Error {
 
     /// Error with a custom message that was preserved.
     #[cfg(feature = "custom-error-messages")]
+    #[cfg_attr(feature = "defmt", defmt(defmt::Debug2Format))]
     CustomErrorWithMessage(heapless::String<64>),
```
Comment (ryan-summers) [inline src/de/mod.rs:79]: As per https://defmt.ferrous-systems.com/format.html#uncompressed-adapters, I think maybe this should be:

```suggestion
    CustomErrorWithMessage(#[cfg_attr(feature = "defmt", defmt(defmt::Debug2Format))] heapless::String<64>),
```

Comment (vhdirk) [reply src/de/mod.rs]: Ah yes, quite true!

**Comment by ryan-summers** [comment]
I think there's one last formatting issue, and one compile issue (don't need a defmt:: prefix). Try it locally with `cargo build --all-features`

**Comment by vhdirk** [comment]
All done!

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -40,9 +40,9 @@ jobs:
       - uses: actions-rs/cargo@v1
         with:
           command: build
-          args: --target=${{ matrix.TARGET }}
+          args: --target=${{ matrix.TARGET }} --all-features
```
Comment (eldruin) [inline .github/workflows/ci.yml:43]:
```suggestion
          args: --target=${{ matrix.TARGET }}
```
The `std` feature cannot be compiled on thumb targets so this cannot be added here.

**Review by eldruin — COMMENTED** [review]
Thank you!

**Comment by vhdirk** [comment]
Sure, done!

**Comment by ryan-summers** [comment]
I think the msrv workflow is still referring to 1.56, sorry for all the tiny details here.

**Comment by eldruin** [comment]
Hmm 1.60 seems not enough. Could you bump it to 1.62.0?

**Review by ryan-summers — APPROVED** [review]
Thanks for working through this, everything's looking good now :)

**Pull request merged**